### DISCUSSION TITLE
[Q&A] Docker部署设置midjourney插件的 PLUGIN_SETTINGS 无效？

### DISCUSSION THREAD

**Discussion by ChenyqThu** [lobehub/lobehub#1102] [Self Hosting]
### 💻 系统环境

Windows

### 📦 部署环境

Docker

### 🌐 浏览器

Chrome

### 🐛 问题描述

Docker部署中增加PLUGIN_SETTINGS，但是实际用户访问时使用midjourney插件，仍然会提示需要输入代理地址。

`-e PLUGIN_SETTINGS='Midjourney:MIDJOURNEY_PROXY_URL=https://example.mj.ink'`
URL地址是正确的（上面为示例），在用户侧单独设置midjourney插件的PROXY_URL，能够正常运行

### 🚦 期望结果

docker部署中设置PLUGIN_SETTINGS，用户可以直接使用插件而无须配置。

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment]
> -e PLUGIN_SETTINGS='Midjourney:MIDJOURNEY_PROXY_URL=https://example.mj.ink'

Midjourney 要小写。

```fish
-e PLUGIN_SETTINGS='midjourney:MIDJOURNEY_PROXY_URL=https://example.mj.ink'
```

**Reply by ChenyqThu** [reply]
换成小写了，还是不行 T.T

**Reply by arvinxx** [reply]
抱歉，刚研究了下发现可能暂时还没法支持这个配置。

其他类型的插件，因为会走一道 `/api/plugin/gateway` 的服务端请求，所以在那个时候，会把这个 env 配置读进去，然后往插件发请求。

但 standalone 的插件其实不会走服务端请求，是直接前端触发的。所以没法从服务端把这个配置读到前端来（或者说也不是没法读，而是发到前端，用户就可能会盗用）。

我得想想怎么解

**Comment by lanchedzy** [comment]
你好，请问在浏览器上设置插件的MIDJOURNEY_PROXY_URL也无效么？
<img width="739" alt="image" src="https://github.com/lobehub/lobe-chat/assets/7939433/5da5878e-3f91-44d3-bc77-18e18baecb01">
我设置完成后，请求依然报同样的问题
